src/Products.jl

```diff
@@ -207,6 +207,86 @@ function locate(lp::LibraryProduct, prefix::Prefix; platform::Platform = platfor
     return nothing
 end
 
+"""
+A `FrameworkProduct` is a  [`Product`](@ref) that encapsulates a macOS Framework.
+It behaves mostly as a [`LibraryProduct`](@ref) and on non-macOS platforms falls back
+to the behavior of `LibraryProduct`, so there is no need to define the products
+differently on non-macOS platforms.
+"""
+struct FrameworkProduct <: Product
+    libraryproduct::LibraryProduct
+
+    """
+        FrameworkProduct(fwnames, varname::Symbol)
+
+    Declares a [`FrameworkProduct`](@ref) that points to a framework located within
+    the `libdir` of the given `Prefix`, with a name containing `fwname` appended with `.framework`.  As an
+    example, given that `libdir(prefix)` is equal to `usr/lib`, and `fwname` is
+    equal to `QtCore`, this would be satisfied by the following paths:
+
+        usr/lib/QtCore.framework
+
+    If a `.framework` is not found, the names are passed along to [`LibraryProduct`](@ref) for processing as
+    a normal shared library.
+    """
+    FrameworkProduct(fwname::AbstractString, varname, args...; kwargs...) = FrameworkProduct([fwname], varname, args...; kwargs...)
+    function FrameworkProduct(fwnames::Vector{<:AbstractString}, varname::Symbol,
+                            dir_paths::Vector{<:AbstractString}=String[];
+                            kwargs...)
+        # If some other kind of AbstractString is passed in, convert it
+        return new(LibraryProduct(fwnames, varname, dir_paths; kwargs...))
+    end
+
+    FrameworkProduct(meta_obj::Dict) = new(LibraryProduct(meta_obj))
+end
+
+Base.:(==)(a::FrameworkProduct, b::FrameworkProduct) = (a.libraryproduct == b.libraryproduct)
```

Comment:
Is this necessary?  I though that since `FrameworkProduct` is an immutable struct `==` would basically compare its fields.

Reply:
I think so, because `FrameworkProduct` is not stored as a bits type. In the pushed update this is tested, and it doesn't work without the additional `==` method.